### ISSUE TITLE
google_checks.xml: incorrect indentation error with block statement in switch

### ISSUE THREAD

**Issue by cushon** [checkstyle/checkstyle#6392]
https://checkstyle.org/config_misc.html#Indentation

```java
class T {
  int main(int i) {
    switch (i) {
      case 42:
        {
          int j = i;
          return j;
        }
      default:
        return 01;
    }
  }
}
```

```
$ java -jar checkstyle-8.15-all.jar -c google_checks.xml T.java
Starting audit...
[WARN] /tmp/tmp.XMfnmlPDg1/T.java:5: 'block lcurly' has incorrect indentation level 8, expected level should be 6. [Indentation]
[WARN] /tmp/tmp.XMfnmlPDg1/T.java:5:9: '{' at column 9 should be on the previous line. [LeftCurly]
[WARN] /tmp/tmp.XMfnmlPDg1/T.java:6: 'block' child has incorrect indentation level 10, expected level should be 8. [Indentation]
[WARN] /tmp/tmp.XMfnmlPDg1/T.java:7: 'block' child has incorrect indentation level 10, expected level should be 8. [Indentation]
[WARN] /tmp/tmp.XMfnmlPDg1/T.java:8: 'block rcurly' has incorrect indentation level 8, expected level should be 6. [Indentation]
Audit done.
```

(`google_checks.xml` was downloaded from: https://raw.githubusercontent.com/checkstyle/checkstyle/master/src/main/resources/google_checks.xml)

The warning is incorrect. The Google Java Style Guide [§4.8.4](https://google.github.io/styleguide/javaguide.html#s4.8.4-switch) says:

> After a switch label, there is a line break, and the indentation level is increased +2, exactly as if a block were being opened. The following switch label returns to the previous indentation level, as if a block had been closed.

**Comment by romani** [comment]
>The warning is incorrect. 

Do you mean all warnings? Or some specific?

**Comment by cushon** [comment]
> Do you mean all warnings?

I meant all of them, thanks for clarifying.

**Comment by romani** [comment]
```
<module name="Indentation">
            <property name="basicOffset" value="2"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="2"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="4"/>
            <property name="arrayInitIndent" value="2"/>
        </module>
```

I do not remember exact reason, probably simple change in config should show cases in `it` tests.

**Comment by rnveach** [comment]
Changing the 0 to 2 produces violations on `}` for method definitions and such. It also affects this issue too. `braceAdjustment` makes sure when both braces are indented the same amount when either/both start the line.

To me I see no way to configure the check with the options we currently have for this issue since we treat all `{`s the same basically.

**Comment by romani** [comment]
Fix is merged
